File: dotnet/src/Model/Player/Units/PlayerUnits.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    class PlayerUnits {
        public List<ITile> units {get;set;}
        int counter;

        public PlayerUnits() {
            this.units = new List<ITile>();
            this.counter = 0;
        }

        List<int> getNextBuilder() {
            if (this.units.Count == 0) {
                return new List<int>(){-1, -1};
            }
            else if (this.units.Count == 1 && this.units[0] is PBuilder) {
                return new List<int>(){this.units[0].loc.row, this.units[0].loc.col};
            }
            int stop = this.units.Count;
            for (int i = 0; i < this.units.Count; ++i) {
                counter+= 1;
                if (counter == this.units.Count) {
                    counter = 0;
                }
                if (this.units[counter] is PBuilder) {
                    return new List<int>(){this.units[counter].loc.row, units[counter].loc.col};
                }
            }
            return new List<int>(){-1, -1};
        }

        List<int> getNextWorker() {
            if (this.units.Count == 0) {
                return new List<int>(){-1, -1};
            }
            else if (this.units.Count == 1 && this.units[0] is PWorker) {
                return new List<int>(){this.units[0].loc.row, this.units[0].loc.col};
            }
            int stop = this.units.Count;
            for (int i = 0; i < this.units.Count; ++i) {
                counter+= 1;
                if (counter == this.units.Count) {
                    counter = 0;
                }
                if (this.units[counter] is PWorker) {
                    return new List<int>(){this.units[counter].loc.row, units[counter].loc.col};
                }
            }
            return new List<int>(){-1, -1};
        }

        List<int> getNextSoldier() {
            if (this.units.Count == 0) {
                return new List<int>(){-1, -1};
            }
            else if (this.units.Count == 1 && this.units[0] is PSoldier) {
                return new List<int>(){this.units[0].loc.row, this.units[0].loc.col};
            }
            int stop = this.units.Count;
            for (int i = 0; i < this.units.Count; ++i) {
                counter+= 1;
                if (counter == this.units.Count) {
                    counter = 0;
                }
                if (this.units[counter] is PSoldier) {
                    return new List<int>(){this.units[counter].loc.row, units[counter].loc.col};
                }
            }
            return new List<int>(){-1, -1};
        }
    }
}

File: jvm2/src/Model/Player/Units/PlayerUnits.kt
package Model.Player.Units

import Model.Abstract.ITile

class PlayerUnits() {
    val units = mutableListOf<ITile>()
    private var counter = 0


    fun getNextBuilder() : MutableList<Int> {
        if (units.size == 0) {
            return mutableListOf<Int>(-1, -1)
        }
        else if (units.size == 1 && units[0] is PBuilder) {
            return mutableListOf<Int>(units[0].loc.row, units[0].loc.col)
        }
        val stop  = units.size
        for (i in 0 until stop) {
            counter += 1
            if (counter == units.size) {
                counter = 0
            }
            if (units[counter] is PBuilder) {
                return mutableListOf<Int>(units[counter].loc.row, units[counter].loc.col)
            }
        }
        return mutableListOf<Int>(-1, -1)
    }

    fun getNextWorker() : MutableList<Int> {
        if (units.size == 0) {
            return mutableListOf<Int>(-1, -1)
        }
        else if (units.size == 1 && units[0] is PWorker) {
            return mutableListOf<Int>(units[0].loc.row, units[0].loc.col)
        }
        val stop  = units.size
        for (i in 0 until stop) {
            counter += 1
            if (counter == units.size) {
                counter = 0
            }
            if (units[counter] is PWorker) {
                return mutableListOf<Int>(units[counter].loc.row, units[counter].loc.col)
            }
        }
        return mutableListOf<Int>(-1, -1)
    }

    fun getNextSoldier() : MutableList<Int> {
        if (units.size == 0) {
            return mutableListOf<Int>(-1, -1)
        }
        else if (units.size == 1 && units[0] is PSoldier) {
            return mutableListOf<Int>(units[0].loc.row, units[0].loc.col)
        }
        val stop  = units.size
        for (i in 0 until stop) {
            counter += 1
            if (counter == units.size) {
                counter = 0
            }
            if (units[counter] is PSoldier) {
                return mutableListOf<Int>(units[counter].loc.row, units[counter].loc.col)
            }
        }
        return mutableListOf<Int>(-1, -1)
    }
}